gnovm/pkg/integration/coverage.go

```diff
@@ -0,0 +1,61 @@
+package integration
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/rogpeppe/go-internal/testscript"
+)
+
+var coverageEnv struct {
+	coverdir string
+	once     sync.Once
+}
+
+// SetupCoverage sets up the given testscripts environment for coverage.
+// It will mostly override `GOCOVERDIR` with the target cover directory
+func SetupCoverage(p *testscript.Params, coverdir string) error {
+	var err error
+
+	// We need to have an absolute path here, because current directory
+	// context will change while executing testscripts.
+	if !filepath.IsAbs(coverdir) {
+		abspath, err := filepath.Abs(coverdir)
+		if err != nil {
+			return fmt.Errorf("unable to determine absolute path of %q: %w", coverdir, err)
+		}
+		coverdir = abspath
+
+	}
+
+	// If the given coverage directory doesn't exist, create it
+	info, err := os.Stat(coverdir)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			if mkErr := os.Mkdir(coverdir, 0o755); mkErr != nil {
+				return fmt.Errorf("failed to testscripts coverage dir %q: %w", coverdir, mkErr)
+			}
+		} else {
+			// Handle other potential errors from os.Stat
+			return fmt.Errorf("failed to stat %q: %w", coverdir, err)
+		}
+	} else if !info.IsDir() {
+		return fmt.Errorf("coverage: %q is not a directory", coverdir)
+	}
```

Comment:
looks better thanks !

Reply:
ok, based on original go `-test.gocoverdir` flag, it seems that they don't create target directory if it not exist: 

```
15:16:38 gno/gnovm/cmd/gno  go test  -p 1 -timeout=30m -covermode=atomic -test.gocoverdir=./foo.

[ ... ]

error generating coverage report: output directory "./foo" inaccessible (err: stat ./foo no such file or directory); no coverage data written
```

so I will mimic this behavior and error since it is deeply related with `gocoverdir` argument